File: BinarySearch/Java/Node.java
package BinarySearch.Java;

/**
 * Node class for the Binary Search Tree.
 * Each node contains a value and references to left and right child nodes.
 */
public class Node {
    private int value;
    private Node left;
    private Node right;

    /**
     * Constructor to create a new node with the given value.
     * 
     * @param value The value to be stored in the node
     */
    public Node(int value) {
        this.value = value;
        this.left = null;
        this.right = null;
    }

    /**
     * Get the value stored in this node.
     * 
     * @return The value stored in this node
     */
    public int getValue() {
        return value;
    }

    /**
     * Set the value for this node.
     * 
     * @param value The new value to be stored
     */
    public void setValue(int value) {
        this.value = value;
    }

    /**
     * Get the left child of this node.
     * 
     * @return The left child node
     */
    public Node getLeft() {
        return left;
    }

    /**
     * Set the left child of this node.
     * 
     * @param left The new left child node
     */
    public void setLeft(Node left) {
        this.left = left;
    }

    /**
     * Get the right child of this node.
     * 
     * @return The right child node
     */
    public Node getRight() {
        return right;
    }

    /**
     * Set the right child of this node.
     * 
     * @param right The new right child node
     */
    public void setRight(Node right) {
        this.right = right;
    }
}

File: BinarySearch/Python/node.py
"""
Node class for the Binary Search Tree.
Each node contains a value and references to left and right child nodes.
"""
class Node:
    def __init__(self, value):
        """
        Constructor to create a new node with the given value.
        
        Args:
            value: The value to be stored in the node
        """
        self.value = value
        self.left = None
        self.right = None
    
    def get_value(self):
        """
        Get the value stored in this node.
        
        Returns:
            The value stored in this node
        """
        return self.value
    
    def set_value(self, value):
        """
        Set the value for this node.
        
        Args:
            value: The new value to be stored
        """
        self.value = value
    
    def get_left(self):
        """
        Get the left child of this node.
        
        Returns:
            The left child node
        """
        return self.left
    
    def set_left(self, left):
        """
        Set the left child of this node.
        
        Args:
            left: The new left child node
        """
        self.left = left
    
    def get_right(self):
        """
        Get the right child of this node.
        
        Returns:
            The right child node
        """
        return self.right
    
    def set_right(self, right):
        """
        Set the right child of this node.
        
        Args:
            right: The new right child node
        """
        self.right = right
